[NIFI-2888] Display processor fill color when zoomed in/out.

scottyaslan opened the pull request:
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:
### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
   in the commit message?
- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?
- [ ] Is your initial contribution a single, squashed commit?
### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?
### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
### Note:

Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

mcgilman commented:
Will review

mcgilman commented on nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-processor.js:
```diff
@@ -547,30 +557,64 @@ nf.Processor = (function () {
+
+                    if (!d.permissions.canRead) {
+                        //update the processor icon container
+                        processor.select('rect.processor-icon-container').classed('unauthorized', true);
```
I don't think we should be updating/modifying these elements within the callback for a different element. Because we're already iterating through each matched processor, `d` is already available as `processorData`. Each of these` processor.select` should be brought outside of this callback.

FYI - we're using `.each` in this case because of how the tooltips are implemented. They didn't directly map into the traditional mapping process.

mcgilman commented:
Looks good @scottyaslan. But I think we need to update the Fill Color dialog so the user can see a preview of their color choice with the contrasting color.

scottyaslan commented:
@mcgilman I have updated the fill color dialog processor preview to match what will actually be applied to the element on the canvas. I also updated the label text to contrast the fill color for the labels so please take a look at that while you review this PR. I updated the fill color dialog label preview as well.

moranr commented:
@scottyaslan, @mcgilman – I think the processor preview looks good, showing only what will change. For me, what is odd is the fact that the initial preview is not an accurate view of the current state of the processor. I feel like an initial color should not be selected and either the preview is empty or it shows the default processor. The preview would change once the user selects a color. Overall it's a trivial concern and could be addressed later; I don't think the current interaction will confuse anyone.

I like the color change of the label value text, but the label preview is using a gradient and the border color changes as well. To be accurate I would suggest using the solid color and keep the border gray (`rgba(0,0,0,0.25`). I'd also suggest updating the text in the label preview to say "Label Value" instead of only "Label." This aligns with the wording seen when you _configure_ the label to enter text.

scottyaslan commented:
@moranr I have updated this PR to include the changes you have suggested for the Label fill color preview and wording.

As for the initial preview I would like to get @mcgilman opinion but to me the preview should be just that...a preview of what WILL be applied and not a view of what currently is.

moranr commented:
Thanks @scottyaslan – agree on getting @mcgilman 's opinion too. Perhaps it's just that initial color choice I'm not fond of. It's fine applied to the processor icon, but at a larger scale it's really not that nice ;)

scottyaslan commented:
Could be...I was also thinking about how this maps to the Label preview...like....why do we show any text if the label doesn't have text currently?

mcgilman commented:
Personally, I think if we're going to attempt to draw a preview of the processor block we need to fill in some of the empty space. Can we try putting in the name of the processor and drawing the two blue bars where we would render the In/Out stats?

Also, can we use more of the vertical space for the preview in the color dialog for a label? Currently, it looks like the label preview is the same dimensions as a processor.

mcgilman commented:
Thanks @scottyaslan! This has been merged to master.
